hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AbstractMessageTask.java

```diff
@@ -89,23 +90,47 @@ public void run() {
         try {
             if (endpoint == null) {
                 handleMissingEndpoint();
-                return;
-            }
-            // process message
-            if (!node.joined()) {
-                throw new HazelcastInstanceNotActiveException("Hazelcast instance is not ready yet!");
+            } else if (isAuthenticatorMessage()) {
+                initializeAndRunMessage();
+            } else if (!endpoint.isAuthenticated()) {
+                handleAuthenticationFailure();
+            } else {
+                initializeAndRunMessage();
             }
-            Credentials credentials = endpoint.getCredentials();
-            interceptBefore(credentials);
-            checkPermissions(endpoint);
-            processMessage();
-            interceptAfter(credentials);
+
         } catch (Throwable e) {
             logProcessingFailure(e);
             handleProcessingFailure(e);
         }
     }
 
+    protected boolean isAuthenticatorMessage() {
+        return false;
+    }
+
+    private void initializeAndRunMessage() {
```

Comment:
Perhaps a more obvious name would be 'initialize' or 'initializeAndProcessMessage'. Otherwise you have run and process as description what you are doing.